src/core/libmaven/peakdetector.cpp

```diff
@@ -156,15 +156,30 @@ void PeakDetector::identifyFeatures(const vector<Compound*>& identificationSet)
     if (identificationSet.empty())
         return;
 
+    vector<Compound*> compoundsWithRawMzOnly;
+    vector<Compound*> compoundsWithNeutralMassOnly;
+    for (auto compound : identificationSet) {
+        if (compound->formula().empty() && compound->neutralMass() > 0.0f) {
+            compoundsWithNeutralMassOnly.push_back(compound);
+        } else if (compound->formula().empty() && compound->mz() > 0.0f) {
+            compoundsWithRawMzOnly.push_back(compound);
+        }
+    }
+
     sendBoostSignal("Preparing libraries for identification…", 0, 0);
     MassSlicer massSlicer(_mavenParameters);
     if (_mavenParameters->pullIsotopesFlag && _mavenParameters->searchAdducts) {
         massSlicer.generateIsotopeSlices(identificationSet);
         massSlicer.generateAdductSlices(identificationSet, true, false);
+        massSlicer.generateCompoundSlices(compoundsWithNeutralMassOnly, false);
+        massSlicer.generateCompoundSlices(compoundsWithRawMzOnly, false);
     } else if (_mavenParameters->pullIsotopesFlag) {
         massSlicer.generateIsotopeSlices(identificationSet);
+        massSlicer.generateCompoundSlices(compoundsWithNeutralMassOnly, false);
+        massSlicer.generateCompoundSlices(compoundsWithRawMzOnly, false);
     } else if (_mavenParameters->searchAdducts) {
         massSlicer.generateAdductSlices(identificationSet);
+        massSlicer.generateCompoundSlices(compoundsWithRawMzOnly, false);
```

Comment:
Why not compounds with neutral mass added here?

Reply:
Right, that was intentional. For isotopes, at least formula is required, but for adducts, either formula or neutral mass is sufficient for calculating adduct mass. So compounds that have formula or neutral mass do not need any special treatment here.